### PR TITLE
feat: support experiments overrides for A/b tests

### PR DESCRIPTION
**Pull request by MounirDhahri**
This PR resolves [] 

### Description

This PR is an effort to improve our experiments infra and make testing different variants with different payloads easier. This PR adds the following:
- Custom variants overrides + custom payload overrides (two different entities because both can live separately)

https://github.com/artsy/eigen/assets/11945712/0bda209f-4425-4a63-90e4-9b3d67503de8

- Since the fields are case-sensitive, I added suggestions for the payload for each experiment for a better UX.

<Image src="https://github.com/artsy/eigen/assets/11945712/f91677e3-b58f-4d93-a7a9-db1ee344e1a1" width="200" />

- Allow users to reset all overrides in one go, thanks for the suggestion @olerichter00 

https://github.com/artsy/eigen/assets/11945712/328fcb95-a7b3-4350-9d40-1d947a200123

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- support experiments overrides in dev menu - mounir

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Review by gkartalis — COMMENTED** [review]
omg

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Cross-platform user-facing changes (support experiments overrides in dev menu - mounir)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 276f7219cd0afd8da343a1c954bcaf3426730429
</p>

File: src/app/system/devTools/DevMenu/Components/Experiments.tsx [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
+import { EXPERIMENT_NAME, experiments } from "app/utils/experiments/experiments"
 
 export const Experiments: React.FC<{}> = () => {
+  const {} = GlobalStore.actions.artsyPrefs.experiments
+
   return (
     <Flex mx={2}>
-      <Expandable label="Experiments" expanded={false}>
+      <Expandable label="Experiments" expanded>
+        {/* <Expandable label="Experiments" expanded={false}> */}
```
Comment (gkartalis) [inline src/app/system/devTools/DevMenu/Components/Experiments.tsx:13]: 👀

**Review by olerichter00 — DISMISSED** [review]
Great idea! That will be very helpful! 🌟 🌟 🌟 

What do you think about adding a "Remove all experiment overrides" button that appears when there are overrides? It could be good to immediately see that there are overrides and to have an easy way to remove all of them to avoid confusion when testing.

We already have this button for feature flags.

**Comment by MounirDhahri** [comment]
That's a nice idea @olerichter00 I will add it

**Comment by MounirDhahri** [comment]
Done @olerichter00 

https://github.com/artsy/eigen/assets/11945712/229241b0-3813-4c0a-bc34-774e2992c2e4

**Review by gkartalis — APPROVED** [review]
huge props for this effort 👏👏👏👏👏👏👏👏

**Pull request merged**